sphinx/ext/intersphinx.py

```diff
@@ -224,8 +224,7 @@ def fetch_inventory(app, uri, inv):
                     uri = path.dirname(newinv)
         with f:
             try:
-                join = localuri and path.join or posixpath.join
-                invdata = read_inventory(f, uri, join)
+                invdata = read_inventory(f, uri)
```

Comment:
This came from PR #3619, there's a comment there about not using `posixpath.join` to join filesystem paths on windows. However it appears to me that this `join` is only used for generating the paths that will be injected into the output, which I think should always be posix/URL-type paths regardless of the OS and if the files are local or remote (the browser or other program should convert them back I think).